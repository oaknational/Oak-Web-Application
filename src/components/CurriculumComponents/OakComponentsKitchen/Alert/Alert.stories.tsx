import { OakThemeProvider, oakDefaultTheme } from "@oaknational/oak-components";
import { Meta, StoryObj } from "@storybook/nextjs";

import Component from ".";

const meta: Meta<typeof Component> = {
  component: Component,
  argTypes: {},
};

export default meta;
type Story = StoryObj<typeof Component>;

const TestComponent = () => {
  return (
    <OakThemeProvider theme={oakDefaultTheme}>
      <Component type="info" message="Info example" />
      <Component type="neutral" message="Neutral example" />
      <Component type="success" message="Success example" />
      <Component type="alert" message="Alert example" />
      <Component type="error" message="Error example" />
    </OakThemeProvider>
  );
};

export const Alert: Story = {
  render: TestComponent,
};
