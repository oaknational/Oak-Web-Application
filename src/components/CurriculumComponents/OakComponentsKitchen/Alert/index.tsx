import {
  OakBox,
  OakFlex,
  OakFlexProps,
  OakIcon,
  OakIconName,
  OakIconProps,
} from "@oaknational/oak-components";

type AlertType = "info" | "neutral" | "success" | "alert" | "error";

const alertTypes: Record<
  AlertType,
  {
    icon: OakIconName;
    iconColorFilter: OakIconProps["$colorFilter"];
    backgroundColour: OakFlexProps["$background"];
    borderColour: OakFlexProps["$borderColor"];
  }
> = {
  info: {
    icon: "info",
    iconColorFilter: "black",
    backgroundColour: "lavender30",
    borderColour: "lavender",
  },
  neutral: {
    icon: "info",
    iconColorFilter: "black",
    backgroundColour: "grey20",
    borderColour: "grey40",
  },
  success: {
    icon: "success",
    iconColorFilter: "black",
    backgroundColour: "mint30",
    borderColour: "mint110",
  },
  alert: {
    icon: "warning",
    iconColorFilter: "black",
    backgroundColour: "lemon30",
    borderColour: "lemon50",
  },
  error: {
    icon: "error",
    iconColorFilter: "red",
    backgroundColour: "red30",
    borderColour: "red",
  },
} as const;

type AlertProps = {
  icon?: OakIconName;
  type?: AlertType;
  message: string;
} & OakFlexProps;

export default function Alert(props: AlertProps) {
  const { type = "info", icon, message, ...rest } = props;
  const iconResult = icon ?? alertTypes[type].icon;
  const iconColorFilterResult = alertTypes[type]?.iconColorFilter;

  return (
    <OakFlex
      data-testid="oak-inline-banner"
      $background={alertTypes[type]?.backgroundColour}
      $pa={"inner-padding-m"}
      $borderRadius={"border-radius-m2"}
      $borderStyle={"solid"}
      $borderColor={alertTypes[type]?.borderColour}
      $flexDirection={"row"}
      $gap="space-between-xs"
      $alignItems={"center"}
      {...rest}
    >
      <OakBox>
        <OakIcon
          iconName={iconResult}
          $colorFilter={iconColorFilterResult}
          $width="all-spacing-7"
          $height="all-spacing-7"
          data-testid="inline-banner-icon"
        />
      </OakBox>
      <OakBox $font={"body-2"} data-testid="inline-banner-message">
        {message}
      </OakBox>
    </OakFlex>
  );
}
