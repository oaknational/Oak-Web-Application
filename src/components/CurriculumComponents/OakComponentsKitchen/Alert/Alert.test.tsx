import Alert from ".";

import renderWithTheme from "@/__tests__/__helpers__/renderWithTheme";

describe("Fieldset", () => {
  test("all options", async () => {
    const { baseElement } = renderWithTheme(
      <>
        <Alert type="info" message="Info example" />
        <Alert type="neutral" message="Neutral example" />
        <Alert type="success" message="Success example" />
        <Alert type="alert" message="Alert example" />
        <Alert type="error" message="Error example" />
      </>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  test("default options", () => {
    const { baseElement } = renderWithTheme(
      <Alert message="Default example" />,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
