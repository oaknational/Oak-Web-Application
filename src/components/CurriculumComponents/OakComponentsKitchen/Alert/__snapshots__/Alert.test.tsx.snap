// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Fieldset > all options 1`] = `
<body>
  <div>
    <div
      class="sc-fqkvVR sc-gsFSXq hVjNbq gXxntV"
      data-testid="oak-inline-banner"
    >
      <div
        class="sc-fqkvVR dzjDLT"
      >
        <div
          class="sc-fqkvVR gBNXQv"
          data-testid="inline-banner-icon"
        >
          <img
            alt="info"
            class="sc-dcJsrY beWtwI"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            src="https://next_public_oak_assets_host/NEXT_PUBLIC_OAK_ASSETS_PATH/v1709052869/icons/Icon_Info_vsx3xi.svg"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
      <div
        class="sc-fqkvVR ZQNWU"
        data-testid="inline-banner-message"
      >
        Info example
      </div>
    </div>
    <div
      class="sc-fqkvVR sc-gsFSXq gxFeVr gXxntV"
      data-testid="oak-inline-banner"
    >
      <div
        class="sc-fqkvVR dzjDLT"
      >
        <div
          class="sc-fqkvVR gBNXQv"
          data-testid="inline-banner-icon"
        >
          <img
            alt="info"
            class="sc-dcJsrY beWtwI"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            src="https://next_public_oak_assets_host/NEXT_PUBLIC_OAK_ASSETS_PATH/v1709052869/icons/Icon_Info_vsx3xi.svg"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
      <div
        class="sc-fqkvVR ZQNWU"
        data-testid="inline-banner-message"
      >
        Neutral example
      </div>
    </div>
    <div
      class="sc-fqkvVR sc-gsFSXq llcuhN gXxntV"
      data-testid="oak-inline-banner"
    >
      <div
        class="sc-fqkvVR dzjDLT"
      >
        <div
          class="sc-fqkvVR gBNXQv"
          data-testid="inline-banner-icon"
        >
          <img
            alt="success"
            class="sc-dcJsrY beWtwI"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            src="https://next_public_oak_assets_host/NEXT_PUBLIC_OAK_ASSETS_PATH/v1699895534/icons/Icon-Success_aiiprx.svg"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
      <div
        class="sc-fqkvVR ZQNWU"
        data-testid="inline-banner-message"
      >
        Success example
      </div>
    </div>
    <div
      class="sc-fqkvVR sc-gsFSXq cmcFkU gXxntV"
      data-testid="oak-inline-banner"
    >
      <div
        class="sc-fqkvVR dzjDLT"
      >
        <div
          class="sc-fqkvVR gBNXQv"
          data-testid="inline-banner-icon"
        >
          <img
            alt="warning"
            class="sc-dcJsrY beWtwI"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            src="https://next_public_oak_assets_host/NEXT_PUBLIC_OAK_ASSETS_PATH/v1704901279/icons/zzszodmk7fvxm9xzzg9s.svg"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
      <div
        class="sc-fqkvVR ZQNWU"
        data-testid="inline-banner-message"
      >
        Alert example
      </div>
    </div>
    <div
      class="sc-fqkvVR sc-gsFSXq kBBbOQ gXxntV"
      data-testid="oak-inline-banner"
    >
      <div
        class="sc-fqkvVR dzjDLT"
      >
        <div
          class="sc-fqkvVR gBNXQv"
          data-testid="inline-banner-icon"
        >
          <img
            alt="error"
            class="sc-dcJsrY cGPxx"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            src="https://next_public_oak_assets_host/NEXT_PUBLIC_OAK_ASSETS_PATH/v1699895534/icons/Icon-Error_r25aza.svg"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
      <div
        class="sc-fqkvVR ZQNWU"
        data-testid="inline-banner-message"
      >
        Error example
      </div>
    </div>
  </div>
</body>
`;

exports[`Fieldset > default options 1`] = `
<body>
  <div>
    <div
      class="sc-fqkvVR sc-gsFSXq hVjNbq gXxntV"
      data-testid="oak-inline-banner"
    >
      <div
        class="sc-fqkvVR dzjDLT"
      >
        <div
          class="sc-fqkvVR gBNXQv"
          data-testid="inline-banner-icon"
        >
          <img
            alt="info"
            class="sc-dcJsrY beWtwI"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            src="https://next_public_oak_assets_host/NEXT_PUBLIC_OAK_ASSETS_PATH/v1709052869/icons/Icon_Info_vsx3xi.svg"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
      <div
        class="sc-fqkvVR ZQNWU"
        data-testid="inline-banner-message"
      >
        Default example
      </div>
    </div>
  </div>
</body>
`;

exports[`Fieldset all options 1`] = `
<body>
  <div>
    <div
      class="sc-fqkvVR sc-gsFSXq hVjNbq gXxntV"
      data-testid="oak-inline-banner"
    >
      <div
        class="sc-fqkvVR dzjDLT"
      >
        <div
          class="sc-fqkvVR gBNXQv"
          data-testid="inline-banner-icon"
        >
          <img
            alt="info"
            class="sc-dcJsrY beWtwI"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            src="https://next_public_oak_assets_host/NEXT_PUBLIC_OAK_ASSETS_PATH/v1709052869/icons/Icon_Info_vsx3xi.svg"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
      <div
        class="sc-fqkvVR ZQNWU"
        data-testid="inline-banner-message"
      >
        Info example
      </div>
    </div>
    <div
      class="sc-fqkvVR sc-gsFSXq gxFeVr gXxntV"
      data-testid="oak-inline-banner"
    >
      <div
        class="sc-fqkvVR dzjDLT"
      >
        <div
          class="sc-fqkvVR gBNXQv"
          data-testid="inline-banner-icon"
        >
          <img
            alt="info"
            class="sc-dcJsrY beWtwI"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            src="https://next_public_oak_assets_host/NEXT_PUBLIC_OAK_ASSETS_PATH/v1709052869/icons/Icon_Info_vsx3xi.svg"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
      <div
        class="sc-fqkvVR ZQNWU"
        data-testid="inline-banner-message"
      >
        Neutral example
      </div>
    </div>
    <div
      class="sc-fqkvVR sc-gsFSXq llcuhN gXxntV"
      data-testid="oak-inline-banner"
    >
      <div
        class="sc-fqkvVR dzjDLT"
      >
        <div
          class="sc-fqkvVR gBNXQv"
          data-testid="inline-banner-icon"
        >
          <img
            alt="success"
            class="sc-dcJsrY beWtwI"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            src="https://next_public_oak_assets_host/NEXT_PUBLIC_OAK_ASSETS_PATH/v1699895534/icons/Icon-Success_aiiprx.svg"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
      <div
        class="sc-fqkvVR ZQNWU"
        data-testid="inline-banner-message"
      >
        Success example
      </div>
    </div>
    <div
      class="sc-fqkvVR sc-gsFSXq cmcFkU gXxntV"
      data-testid="oak-inline-banner"
    >
      <div
        class="sc-fqkvVR dzjDLT"
      >
        <div
          class="sc-fqkvVR gBNXQv"
          data-testid="inline-banner-icon"
        >
          <img
            alt="warning"
            class="sc-dcJsrY beWtwI"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            src="https://next_public_oak_assets_host/NEXT_PUBLIC_OAK_ASSETS_PATH/v1704901279/icons/zzszodmk7fvxm9xzzg9s.svg"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
      <div
        class="sc-fqkvVR ZQNWU"
        data-testid="inline-banner-message"
      >
        Alert example
      </div>
    </div>
    <div
      class="sc-fqkvVR sc-gsFSXq kBBbOQ gXxntV"
      data-testid="oak-inline-banner"
    >
      <div
        class="sc-fqkvVR dzjDLT"
      >
        <div
          class="sc-fqkvVR gBNXQv"
          data-testid="inline-banner-icon"
        >
          <img
            alt="error"
            class="sc-dcJsrY cGPxx"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            src="https://next_public_oak_assets_host/NEXT_PUBLIC_OAK_ASSETS_PATH/v1699895534/icons/Icon-Error_r25aza.svg"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
      <div
        class="sc-fqkvVR ZQNWU"
        data-testid="inline-banner-message"
      >
        Error example
      </div>
    </div>
  </div>
</body>
`;

exports[`Fieldset default options 1`] = `
<body>
  <div>
    <div
      class="sc-fqkvVR sc-gsFSXq hVjNbq gXxntV"
      data-testid="oak-inline-banner"
    >
      <div
        class="sc-fqkvVR dzjDLT"
      >
        <div
          class="sc-fqkvVR gBNXQv"
          data-testid="inline-banner-icon"
        >
          <img
            alt="info"
            class="sc-dcJsrY beWtwI"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            src="https://next_public_oak_assets_host/NEXT_PUBLIC_OAK_ASSETS_PATH/v1709052869/icons/Icon_Info_vsx3xi.svg"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
      <div
        class="sc-fqkvVR ZQNWU"
        data-testid="inline-banner-message"
      >
        Default example
      </div>
    </div>
  </div>
</body>
`;
